### PR TITLE
Cache downloads before starting build. (fixes #59)

### LINKED ISSUE #59: Cache input source and binaries
Rather than downloading the Base/Corepops and packages each time after a clean, we can instead download them once into a single directory, then build from them.
As part of this, we should add a `deepclean` target to remove these downloads.

### PR DESCRIPTION
**Pull request by willprice**
Why? This separates the process of downloading the dependencies and
then building them. This helps support packaging efforts as we can just
download the dependencies and tar up the directory ready to be built.

Steve, you probably object to the use of `cp`, which I will remove and replace with the `tar` pattern, if you can convince me why it is problematic ;)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -351,18 +354,22 @@ _build/poplog_base/pop/pop/newpop.psv: _build/Stage1.proxy
         && (cd $$popsys; $$popsys/corepop %nort ../lib/lib/mkimage.p -entrymain ./newpop.psv ../lib/lib/newpop.p)
 
 # This target ensures that we have an unpacked base system with a valid corepop file.
-_build/Corepops.proxy: _build/Base.proxy _build/Corepops.Downloaded.proxy
+_build/Corepops.proxy: _build/Base.proxy _download/Corepops.Downloaded.proxy
+	cp -r _download/Corepops _build/
```
Comment (sfkleach) [inline Makefile]: Should be `cp -rpP` I think. The `-p` is to preserve permissions and important to protect against weird umask settings. And `-P` is just a good idea but actually doesn't matter as there are no symlinks in the `Corepops` repo.

File: Makefile [hunk 2]
```diff
@@ -351,18 +354,22 @@ _build/poplog_base/pop/pop/newpop.psv: _build/Stage1.proxy
...
+_download/Corepops.Downloaded.proxy:
+	mkdir -p _download/Corepops
+	curl -LsS $(COREPOPS_TARBALL_URL) | ( cd _download/Corepops; tar zxf - --strip-components=1 )
 	touch $@
 
-_build/Base.proxy: _build/Base.Downloaded.proxy
+_build/Base.proxy: _download/Base.Downloaded.proxy
+	mkdir -p _build
+	cp -r _download/Base/ _build/
+	ls -l _build
```
Comment (sfkleach) [inline Makefile]: Debugging? Should be removed.

**Comment by sfkleach** [comment]
Really nice treatment of a delicate change. Super job. I have added a couple of minor comments that I think need sorting out.

File: Makefile (hunk 2 above)
Comment (willprice) [reply Makefile]: Done.

File: Makefile (hunk 1 above)
Comment (willprice) [reply Makefile]: Done.

**Pull request merged**